core/services/functions/listener.go

```diff
@@ -488,3 +508,18 @@ func (l *FunctionsListener) timeoutRequests() {
 		}
 	}
 }
+
+func (l *FunctionsListener) reportSourceCodeDomains(requestId [32]byte, domains []string) {
+	r := &telem.FunctionsRequest{
+		RequestId: requestId[:],
+		Domains:   domains,
+	}
+
+	bytes, err := proto.Marshal(r)
+	if err != nil {
+		l.logger.Warnw("telem.FunctionsRequest marshal error", "err", err)
+	} else {
+		l.logger.Debugw("Reporting domains", "requestID", formatRequestId(requestId), "domains", domains)
+		l.urlsMonEndpoint.SendLog(bytes)
```

Comment:
Is this non-blocking? Doesn't take Context so I'm assuming so.

Reply:
They use `context.Background` internally :(
I will provide a feedback for them. And then they schedule this send on a worker. So yes, it should be non-blocking.